value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -2395,9 +2433,49 @@ this.[n] = [valueFromValue v][invokeSuper v].[v.names.get]()[/valueFromValue];
 [/if]
     [generateAfterConstruction type false]
   }
+
+[-- Additional constructors for Optional types --]
+[if type.hasOptionalConstructorArguments]
+  [if type.factoryOf.new]
+  /**
+   * Construct a new immutable {@code [type.name]} instance.
+  [for v in type.constructorArguments]
+   * @param [v.name] The value for the {@code [v.name]} attribute[if v.nullable], can be {@code null}[/if]
+  [/for]
+   */
+  [eachLine type.constructorAnnotations]
+  [eachLine type.constructorInjectedAnnotations]
+  [suppressAnyCovariantCastForOptional type.constructorArguments]
+  public [type.typeImmutable.simple]([output.linesShortable][for v in type.constructorArguments][if not for.first],[/if]
+      [v.constructorParameterAnnotations][v.atNullability][if v.isOptionalType][unwrappedOptionalType v][else][constructorAcceptType v][/if] [v.name][/for][/output.linesShortable]) {
+  [else]
+  [suppressAnyCovariantCastForOptional type.constructorArguments]
+  private [type.typeImmutable.simple]([output.linesShortable][for v in type.constructorArguments][if not for.first],[/if]
+      [v.atNullability][if v.isOptionalType][unwrappedOptionalType v][else][constructorAcceptType v][/if] [v.name][/for][/output.linesShortable]) {
```

Comment:
need to add `@Nullable` (or equivalent) where it is substitute for `Optional`. Here I assume we only propagate nullable if it's declared nullable, but I might be wrong here. Have you checked those nullable annotations?

Reply:
As far as i saw this behaves as expected according to the parameter `optionalAcceptNullable` if it is true `@Nullable` is added otherwise it isn't.
If I also correct and add `Objects.requireNonNull()` (see first comment) it would behave correctly in my opinion.
The `atNullable` is called inside of `unwrappedOptionalType`.